tests/integration/helm/upgrade/util.go

```diff
@@ -0,0 +1,188 @@
+// +build integ
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helmupgrade
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/cluster"
+	kubecluster "istio.io/istio/pkg/test/framework/components/cluster/kube"
+	"istio.io/istio/pkg/test/framework/image"
+	"istio.io/istio/pkg/test/helm"
+	kubetest "istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/retry"
+	helmtest "istio.io/istio/tests/integration/helm"
+	"istio.io/istio/tests/util/sanitycheck"
+)
+
+const (
+	gcrHub = "gcr.io/istio-release"
+
+	defaultValues = `
+global:
+  hub: %s
+  tag: %s
+`
+)
+
+// previousChartPath is path of Helm charts for previous Istio deployments.
+var previousChartPath = filepath.Join(env.IstioSrc, "tests/integration/helm/testdata/")
+
+// upgradeCharts upgrades Istio using Helm charts with the provided
+// override values file to the latest charts in $ISTIO_SRC/manifests
+func UpgradeCharts(ctx framework.TestContext, h *helm.Helm, overrideValuesFile string) {
```

Comment:
Except for `PerformUpgradeFunc` all of these functions remained the same, but were moved into this file.

Even `PerformUpgradeFunc` is copied from the `Run()` step in each test to facilitate better testing.